tools/compare.go

```diff
@@ -17,64 +17,137 @@ package tools
+	`),
+	Args: cobra.ExactArgs(3),
+	RunE: compareBlocksE,
+	Example: cli.Dedent(`
+		# Run over full block range
+		fireeth tools compare-blocks sf_bundle/ cs_bundle/ 100-200
+
+		# Run over specific block range (inclusive/inclusive)
+		fireeth tools compare-blocks --diff=true sf_bundle/ cs_bundle/ 100-200
+	`),
 }
 
 func init() {
 	Cmd.AddCommand(compareBlocksCmd)
+	compareBlocksCmd.PersistentFlags().Bool("diff", false, "A flag to check for differences over small range")
 }
 
 func compareBlocksE(cmd *cobra.Command, args []string) error {
+	fmt.Printf("\n-----starting comparison-----\n")
 	ctx := cmd.Context()
 
-	storeADef := args[0]
-	storeBDef := args[1]
+	storeExpectedDef := args[0]
+	storeReceivedDef := args[1]
+	blockRange := block.ParseRange(args[2])
 
-	blockNum, err := strconv.ParseUint(args[2], 10, 64)
+	storeExpected, err := dstore.NewDBinStore(storeExpectedDef)
 	if err != nil {
-		return fmt.Errorf("unable to parse block number %q: %w", args[0], err)
+		return fmt.Errorf("unable to create store at path %q: %w", storeExpectedDef, err)
 	}
 
-	storeA, err := dstore.NewDBinStore(storeADef)
+	storeReceived, err := dstore.NewDBinStore(storeReceivedDef)
 	if err != nil {
-		return fmt.Errorf("unable to create store at path %q: %w", storeADef, err)
+		return fmt.Errorf("unable to create store at path %q: %w", storeReceivedDef, err)
 	}
 
-	storeB, err := dstore.NewDBinStore(storeBDef)
-	if err != nil {
-		return fmt.Errorf("unable to create store at path %q: %w", storeBDef, err)
-	}
+	blocksExpected := make(map[uint64]*pbeth.Block)
+	blocksReceived := make(map[uint64]*pbeth.Block)
+
+	collectBlocksExpected := func(store dstore.Store, blockMap map[uint64]*pbeth.Block) error {
+		fmt.Printf("collecting expected blocks\n")
+		var files []string
+		err = storeExpected.Walk(ctx, "", func(filename string) (err error) {
+			files = append(files, filename)
+			return nil
+		})
+
+		var toClose []io.ReadCloser
+		defer func() {
+			for i := range toClose {
+				err := toClose[i].Close()
+				if err != nil {
+					return
+				}
+			}
+		}()
 
-	blocksA := make(map[string]*pbeth.Block)
-	blocksB := make(map[string]*pbeth.Block)
+		for _, filepath := range files {
+			reader, err := store.OpenObject(ctx, filepath)
+			if err != nil {
+				return err
+			}
+			toClose = append(toClose, reader)
 
-	filePrefix := fmt.Sprintf("%010d", blockNum)
-	collectBlocks := func(store dstore.Store, blockMap map[string]*pbeth.Block) error {
+			blockReader, err := bstream.GetBlockReaderFactory.New(reader)
+			if err != nil {
+				return err
+			}
+
+			for {
+				curBlock, err := blockReader.Read()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return err
+				}
+				if blockRange.Contains(curBlock.Number) {
+					blockMap[curBlock.Number] = curBlock.ToProtocol().(*pbeth.Block)
+					if uint64(len(blockMap)) >= blockRange.Len() {
+						break
+					}
+				}
+			}
+		}
+
+		fmt.Printf("finished collected expected blocks\n")
```

Comment:
No need for a `Printf` here because there is no argument. Pretty sure it shows in yellow in your IDE.